### DISCUSSION TITLE
Pre-commit template

### DISCUSSION THREAD

**Discussion by MitchellBerend** [chshersh/tool-sync#77] [Q&A] [answered]
I added a personal pre-commit since I often forget to run `cargo test` and `cargo fmt`. Is this something we want to add to help others start on tasks?

---
My current (super simple) pre-commit:
```bash
#! /bin/bash

cargo test || exit 1
cargo fmt --all --check
```

**Comment by chshersh** [comment, marked as answer]
That would be a nice improvement indeed 👍🏻 

I suggest using [Pre Commit hooks](https://pre-commit.com/) which is a common tool for this task. So we can create the `.pre-commit-config.yaml` file to this repository and everyone who uses `pre-commit` would benefit from it.

`clippy` would be another useful tool to add in the pre-commit hooks.

Feel free to create a `dx` issue or even a pull request w.r.t. adding pre-commit hooks 🚢
